src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs

```diff
@@ -478,11 +459,11 @@ private async Task<IActionResult> ExchangeClientCredentialsGrantType(OpenIddictR
                 }
             }
 
-            var principal = new ClaimsPrincipal(identity);
-            principal.SetScopes(request.GetScopes());
-            principal.SetResources(await GetResourcesAsync(request.GetScopes()));
+            identity.SetScopes(request.GetScopes());
+            identity.SetResources(await GetResourcesAsync(request.GetScopes()));
+            identity.SetDestinations(GetDestinations);
```

Comment:
In the current bits, `ExchangeClientCredentialsGrantType()` doesn't use the `GetDestinations()` helper. This PR changes that so it also uses it, but I added a check in `GetDestinations()` to preserve the existing values if destinations were already attached to a claim.